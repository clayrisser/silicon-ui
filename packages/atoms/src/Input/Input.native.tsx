import React, { FC } from 'react';
import { Item as NativeBaseItem, Input as NativeBaseInput } from 'native-base';
import {
  background,
  border,
  color,
  layout,
  position,
  shadow,
  space,
  typography
} from 'styled-system';
import useColor from '../hooks/useColor';
import useItem from '../hooks/useItem';
import { createStyled } from '../styled';
import {
  InputProps,
  antiForwardInputPropsKeys,
  splitProps
} from './inputProps';

const StyledNativeBaseInput = createStyled<InputProps>(
  NativeBaseInput,
  [background, border, color, layout, position, shadow, space, typography],
  antiForwardInputPropsKeys
);

const Input: FC<InputProps> = (props: InputProps) => {
  const item = useItem();
  const color = useColor(props);
  const {
    customInputProps,
    nativeInputProps,
    nativeItemProps,
    styledInputProps
  } = splitProps({
    ...props,
    color
  });
  const styledNativeBaseInput = (
    <StyledNativeBaseInput
      {...customInputProps}
      {...nativeInputProps}
      {...styledInputProps}
    />
  );
  if (item.hasItemParent) return styledNativeBaseInput;
  return (
    <NativeBaseItem {...nativeItemProps}>
      {styledNativeBaseInput}
    </NativeBaseItem>
  );
};

Input.defaultProps = {
  // fontFamily: 'body',
  // fontWeight: 'body',
  // lineHeight: 'body',
  autoContrast: false,
  backgroundColor: '#FFFFFF00',
  fontSize: 0
};

export default Input;
