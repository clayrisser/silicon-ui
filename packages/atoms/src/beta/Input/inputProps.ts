import { NativeBase } from 'native-base';
import { SxStyleProp } from 'theme-ui';
import { ThemedStyledProps } from 'native-theme-ui';
import { createSplitProps } from '../../util';

export type InputProps = StyledNativeBaseInputProps & CustomInputProps;

export interface StyledNativeBaseInputProps
  extends Omit<ThemedStyledProps<NativeBase.Input>, 'ref'> {}
export type StyledNativeBaseInputSx = Partial<SxStyleProp>;

export interface CustomInputProps {
  placeholder?: string;
}

interface SplitProps {
  styledNativeBaseInputProps: StyledNativeBaseInputProps;
  customInputProps: CustomInputProps;
}

export interface SplitSx {
  styledNativeBaseInputSx: StyledNativeBaseInputProps;
}

export const splitProps = createSplitProps<InputProps, SplitProps, SplitSx>(
  { customInputProps: ['placeholder'] },
  'styledNativeBaseInputProps',
  {},
  'styledNativeBaseInputSx'
);
