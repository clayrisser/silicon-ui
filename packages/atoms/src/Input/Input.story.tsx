import React from 'react';
import {
  withKnobs,
  text,
  select,
  boolean,
  number
} from '@storybook/addon-knobs';
import Input from './Input';
import Wrapper from '../../storybook/Wrapper';
import storiesOf from '../../storybook/storiesOf';
import themes from '../themes';
import withThemesProvider from '../../storybook/withThemesProvider';
// import docs from './Input.docs.mdx';

storiesOf('Input', module)
  .addDecorator(withKnobs)
  .addDecorator(withThemesProvider(themes))
  .addParameters({
    // docs: { page: docs },
    // jest: ['Input.spec.tsx']
  })
  .add('with knobs', () => (
    <Wrapper>
      <Input
        type={select('type', ['text', 'password'], 'text')}
        backgroundColor={select(
          'backgroundColor',
          ['primary', 'secondary', '#ADFF2F', '#E5FFCC', '#CCFFE5'],
          ''
        )}
        disabled={boolean('disabled', false)}
        maxLength={number('maxLength', 10)}
        minLength={number('minLength', 0)}
        required={boolean('required', false)}
        paddingBottom={number('padding-bottom', 0)}
        paddingTop={number('padding-top', 0)}
        paddingLeft={number('padding-left', 0)}
        paddingRight={number('padding-right', 0)}
        label={text('input-label', 'input')}
        color={text('color', 'text')}
        borderWidth={text('borderWidth', '2px')}
        id={text('input-id', '')}
        width={text('width', '70%')}
        ml={0}
        pl={0}
        placeholder={text('placeholder', 'Sample Placeholder')}
        autoContrast={select<'A' | 'AA' | 'AAA'>(
          'autoContrast',
          {
            false: '' as 'A',
            A: 'A',
            AA: 'AA',
            AAA: 'AAA'
          },
          'AA'
        )}
      />
    </Wrapper>
  ));
