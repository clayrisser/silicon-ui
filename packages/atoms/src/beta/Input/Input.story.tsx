import React, { FC } from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs, select, text } from '@storybook/addon-knobs';
import Input from './Input';
import Wrapper from '../../../storybook/Wrapper';
import docs from './docs';
import storiesOf from '../../../storybook/storiesOf';
import withThemeProvider from '../../../storybook/withThemeProvider';

export const InputStoryWithKnobs: FC = () => (
  <Wrapper>
    <Input
      sx={{ border: text('border', '1px solid black') }}
      placeholder="Enter a value"
    />
  </Wrapper>
);

storiesOf('Beta/Input', module)
  .addDecorator(withKnobs)
  .addDecorator(withThemeProvider)
  .addParameters({
    docs: { page: docs }
    // jest: ['Button.test.tsx']
  })
  .add('with knobs', () => <InputStoryWithKnobs />);
