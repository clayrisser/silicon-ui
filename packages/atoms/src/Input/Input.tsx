import React, { FC } from 'react';
import styled, { StyledComponent } from '@emotion/styled';
import {
  background,
  border,
  color,
  compose,
  layout,
  position,
  shadow,
  space,
  typography
} from 'styled-system';
import useColor from '../hooks/useColor';
import { InputProps, DetailedHTMLInputProps, splitProps } from './inputProps';

const HTMLInput: StyledComponent<
  DetailedHTMLInputProps,
  InputProps,
  object
> = styled.input(
  compose(
    background,
    border,
    color,
    layout,
    position,
    shadow,
    space,
    typography
  )
);

const Input: FC<InputProps> = (props: InputProps) => {
  const color = useColor(props);
  const { customInputProps, styledInputProps, nativeItemProps } = splitProps({
    ...props,
    color
  });
  function handleInput(e: any) {
    if (props.onFocus) props.onFocus(e.target.value);
  }
  return (
    <HTMLInput
      {...styledInputProps}
      {...nativeItemProps}
      {...(customInputProps as any)}
      onInput={handleInput}
      onChangeText={handleInput}
    />
  );
};

Input.defaultProps = {
  autoContrast: false,
  backgroundColor: 'inverseText',
  borderRadius: 2,
  borderWidth: 0,
  fontFamily: 'body',
  fontSize: 2,
  fontWeight: 'body',
  lineHeight: 'body',
  paddingBottom: 2,
  paddingLeft: 2,
  paddingRight: 2,
  paddingTop: 2,
  width: '100%'
};

export default Input;
