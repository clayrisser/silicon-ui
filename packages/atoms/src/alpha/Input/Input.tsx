import React, { FC } from 'react';
import styled, { StyledComponent } from '@emotion/styled';
import {
  background,
  border,
  color,
  compose,
  layout,
  position,
  shadow,
  space,
  typography
} from 'styled-system';
import { InputProps, DetailedHTMLInputProps, splitProps } from './inputProps';

const HTMLInput: StyledComponent<
  DetailedHTMLInputProps,
  InputProps,
  object
> = styled.input(
  compose(
    background,
    border,
    color,
    layout,
    position,
    shadow,
    space,
    typography
  )
);

const Input: FC<InputProps> = (props: InputProps) => {
  const { customInputProps, styledInputProps, nativeItemProps } = splitProps(
    props
  );
  function handleInput(e: any) {
    if (props.onChangeText) props.onChangeText(e.target.value);
  }
  return (
    <HTMLInput
      {...styledInputProps}
      {...nativeItemProps}
      {...(customInputProps as any)}
      onChange={handleInput}
      onChangeText={handleInput}
    />
  );
};

Input.defaultProps = {
  autoContrast: false,
  backgroundColor: 'inverseText',
  borderRadius: 0,
  borderWidth: 0,
  fontFamily: 'body',
  fontSize: 0,
  fontWeight: 'body',
  lineHeight: 'body',
  paddingBottom: 0,
  paddingLeft: 0,
  paddingRight: 0,
  paddingTop: 0,
  width: '100%'
};

export default Input;
