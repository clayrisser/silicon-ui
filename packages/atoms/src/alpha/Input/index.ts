import Input from './Input';
import { InputProps } from './inputProps';

export { InputProps };
export default Input;
