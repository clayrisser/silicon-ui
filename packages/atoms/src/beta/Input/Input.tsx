import React, { FC } from 'react';
import { NativeBase, Input as NativeBaseInput } from 'native-base';
import { styled } from 'native-theme-ui';
import { SxStyleProp } from 'theme-ui';
import { InputProps, splitProps } from './inputProps';

const StyledNativeBaseInput = styled<NativeBase.Input>(NativeBaseInput)({});

const defaultSx: SxStyleProp = {
  border: '1px solid black'
};

const Input: FC<InputProps> = (props: InputProps) => {
  const { sx, styledNativeBaseInputProps } = splitProps(props, defaultSx);
  const { placeholder } = props;
  return (
    <StyledNativeBaseInput
      placeholder={placeholder}
      {...styledNativeBaseInputProps}
      sx={sx}
    />
  );
};

Input.defaultProps = {};

export default Input;
