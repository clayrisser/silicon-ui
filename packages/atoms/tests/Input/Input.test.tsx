import React from 'react';
import renderer from 'react-test-renderer';
// import { render, fireEvent } from '@testing-library/react';

import Input from '../../src/Input';

describe('<Input />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Input />', () => {
  it('renders correctly', () => {
    const fontSize = 16;
    const bg = 'white';
    const color = 'black';
    const height = 90;
    const padding = 9;
    const width = 600;
    const border = 1;
    const borderStyle = 'solid';
    const borderColor = 'green';
    const backgroundColor = 'rerd';
    const lineHeight = 2;
    const autoContrast = true;
    const fontWeight = 2;
    const borderWidth = 1;
    const paddingTop = 6;
    const tree = renderer
      .create(
        <Input
          fontSize={fontSize}
          fontWeight={fontWeight}
          borderWidth={borderWidth}
          bg={bg}
          color={color}
          height={height}
          p={padding}
          border={border}
          width={width}
          borderStyle={borderStyle}
          borderColor={borderColor}
          backgroundColor={backgroundColor}
          autoContrast={autoContrast}
          lineHeight={lineHeight}
          paddingTop={paddingTop}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
