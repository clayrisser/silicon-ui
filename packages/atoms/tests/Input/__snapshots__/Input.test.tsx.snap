// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<input
  className="css-119ltm9"
  fontFamily="body"
  fontSize={0}
  fontWeight="body"
  onChange={[Function]}
  onChangeText={[Function]}
  width="100%"
/>
`;

exports[`<Input /> renders correctly 2`] = `
<input
  className="css-azsq89"
  color="black"
  fontFamily="body"
  fontSize={16}
  fontWeight={2}
  height={90}
  onChange={[Function]}
  onChangeText={[Function]}
  width={600}
/>
`;
